File: app/router/setting.go
package router

import (
	"github.com/aihop/gopanel/app/api"
	"github.com/aihop/gopanel/app/middleware"
	"github.com/aihop/gopanel/constant"
	"github.com/gofiber/fiber/v3"
)

func SettingRouter(r fiber.Router) {
	settingGroup := r.Group("/setting")
	settingGroup.Use(middleware.JWT(constant.UserRoleAdmin))
	{
		settingGroup.Post("/system/info", api.SettingSystemInfo)
		settingGroup.Post("/system/update", api.SettingSystemUpdate)

		settingGroup.Get("/system/version", api.SettingSystemVersion)
		settingGroup.Get("/system/check", api.SettingSystemCheck)
		settingGroup.Post("/system/upgrade", api.SettingSystemUpgrade)
		settingGroup.Get("/system/upgrade/logs", api.SettingSystemUpgradeLogs)
		settingGroup.Post("/system/restart", api.SettingSystemRestart)
		settingGroup.Post("/system/restart/:operation", api.SettingSystemRestart)

		settingGroup.Post("/system/config", api.SettingSystemConfig)
		settingGroup.Post("/system/port", api.SettingSystemPort)

		settingGroup.Post("/system/entrance", api.SettingSystemEntrance)

		settingGroup.Post("/system/clear", api.SettingSystemClearDir)
		settingGroup.Post("/system/baseDir", api.SettingSystemBaseDir)
		settingGroup.Post("/system/apiToken", api.SettingSystemApiTokenUpdate)

	}
}

File: admin/src/api/modules/setting.ts
import http from "@/api"
import type { Setting } from "../interface/setting"

export const UploadFileData = (params: FormData) => {
	return http.upload("/licenses/upload", params)
}

export const getSettingInfo = () => {
	return http.post<Setting.SettingInfo>(`/setting/system/info`)
}

export const settingSystemBaseDirAPI = () => {
	return http.post<string>(`/setting/system/base-dir`)
}

export const updateSetting = (param: Setting.SettingUpdate) => {
	return http.post(`/setting/system/update`, param)
}

// 获取当前版本
export const settingSystemVersionAPI = () => {
	return http.get("/setting/system/version")
}

// 检查更新
export const settingSystemCheckAPI = (params: { lang: string, appBrand: string }) => {
	return http.get("/setting/system/check", params)
}

// 更新
export const settingSystemUpgradeAPI = (params: {
	containerName: string
	currentVersion: string
	targetVersion: string
}) => {
	return http.post("/setting/system/upgrade", params)
}

export const settingSystemConfig = () => {
	return http.post<any>(`/setting/system/config`)
}

export const settingSystemPort = (param: { serverPort: number }) => {
	return http.post<any>(`/setting/system/port`, param)
}

export const settingSystemEntrance = (param: { entrance: string }) => {
	return http.post<any>(`/setting/system/entrance`, param)
}

export const settingSystemClear = (param: { key: "log" | "tmp" | "cache" }) => {
	return http.post<any>(`/setting/system/clear`, param)
}

export const settingSystemApiTokenUpdate = (params: { apiInterfaceStatus: string, apiKey: string }) => {
	return http.post(`/setting/system/api-token`, params)
}

export const settingSystemRestart = (operation: "panel" | "server" = "panel") => {
	return http.post<any>(`/setting/system/restart/${operation}`)
}
